src/pages/Bridge/RedeemForm/index.tsx

```diff
@@ -249,14 +293,21 @@ const RedeemForm = (): JSX.Element | null => {
             return;
           }
 
-          vaultId = getRandomVaultIdWithCapacity(Array.from(updatedVaults || new Map()), wrappedTokenAmount);
+          if (selectVaultManually) {
+            if (!selectedVault) {
+              throw new Error('Specific vault is not selected!');
+            }
+            vaultId = selectedVault[0];
+          } else {
+            vaultId = getRandomVaultIdWithCapacity(Array.from(updatedVaults || new Map()), monetaryWrappedTokenAmount);
+          }
         }
 
         // FIXME: workaround to make premium redeem still possible
         const relevantVaults = new Map<InterbtcPrimitivesVaultId, BitcoinAmount>();
         // FIXME: a bit of a dirty workaround with the capacity
-        relevantVaults.set(vaultId, wrappedTokenAmount.mul(2));
-        const result = await window.bridge.redeem.request(wrappedTokenAmount, data[BTC_ADDRESS], vaultId);
+        relevantVaults.set(vaultId, monetaryWrappedTokenAmount.mul(2));
+        const result = await window.bridge.redeem.request(monetaryWrappedTokenAmount, data[BTC_ADDRESS], vaultId);
```

Comment:
I found this `FIXME` logic was not used anywhere. I was going to delete it but wanted to confirm first. Do you have any idea if we really need this or not?

Reply:
For now, we discussed to leave it as it is. I opened https://github.com/interlay/interbtc-ui/issues/748.